specs/vm/instruction_set.md

```diff
@@ -1386,6 +1387,55 @@ Cease VM execution and revert script effects. After a revert:
 1. All [OutputContract](../protocol/tx_format.md#outputcontract) outputs will have the same `balanceRoot` and `stateRoot` as on initialization.
 1. All [OutputVariable](../protocol/tx_format.md#outputvariable) outputs will have `to`, `amount`, and `asset_id` of zero.
 
+### SMO: Send message to output
+
+|             |                                                                                     |
+|-------------|-------------------------------------------------------------------------------------|
+| Description | Send a message to recipient address `MEM[$rA, 32]` with call abi `MEM[$rA + 32, $rB]` and `$rD` coins, with output `$rC`. |
+| Operation   | ```outputmessage(MEM[$fp, 32], MEM[$rA, 32], MEM[$rA + 32, $rB], $rD, $rC);```      |
+| Syntax      | `smo $rA, $rB, $rC, $rD`                                                            |
+| Encoding    | `0x00 rA rB rC rD`                                                                  |
+| Notes       |                                                                                     |
+
+Given helper `balanceOfStart(asset_id: byte[32]) -> uint32` which returns the memory address of the remaining free balance of `asset_id`, or panics if `asset_id` has no free balance remaining.
+
+Panic if:
+
+- `$rA + 32` overflows
+- `$rA + $rB + 32` overflows
+- `$rA + 32 > VM_MAX_RAM`
+- `$rA + $rB + 32 > VM_MAX_RAM`
+- `$rB > MEM_MAX_ACCESS_SIZE`
+- `$rB > 65535`
```

Comment:
Where does `65535` come from? Is it to fit into a `u16`? If so, then that should be captured by `MEM_MAX_ACCESS_SIZE`.

Reply:
MEM_MAX_ACCESS_SIZE is a u64. Is this incorrect in the spec?

Reply:
Yes, first because that's just the type of the value, second because of https://github.com/FuelLabs/fuel-specs/issues/298

Reply:
Ok, well maybe we should keep them different in case one changes but not the other. I'm talking about defining a `MESSAGE_MAX_DATA_SIZE` similar to `CONTRACT_MAX_SIZE`

Reply:
Yes, it could be `MESSAGE_MAX_DATA_LENGTH` described here https://github.com/FuelLabs/fuel-specs/pull/318#discussion_r895074789

Reply:
Ah nvm, got it, one's for input and the other's for the receipt.